index.js

```diff
@@ -74,25 +75,29 @@ function compression (options) {
 
     // proxy
 
-    res.write = function write (chunk, encoding) {
+    res.write = function write (chunk, encoding, cb) {
       if (ended) {
         return false
       }
 
+      cb = (res._write.length === 3) ? cb : null
```

Comment:
I don't think this is going yo be robust, for example, I see the following pretty frequently:

    res.write = function write () {
      // done
      return _write.apply(this, arguments)
    }

This is a very common pattern to wrap some function, but the length of the function ends up being zero, which means this module won't send the callback (and the wrapping may only be for certain requests, complicating debugging, which would point to "when I use compression, my callbacks sometimes do not work. Removing compression works fine").

Thoughts?